### PR TITLE
Hide Exact Data on Front-end

### PR DESCRIPTION
**Pull request by owensimpson**
### Notes
- Added ``PublicUser`` type which is the ``User`` type without exact data. 
- Generated Point of Interest (POI) data using the Mapbox API.
- Ensured only POI data is visible on the front end.
- Updated seed.ts to support updated data and fixed bugs. 

### Testing
- Visually tested using ``prisma studio``, using the developer tools, and using the app. 

### Issues
- [Updating all instances of location usage/querying](https://trello.com/c/9qvQQxZK)

### CODE REVIEW

File: prisma/seed.ts [hunk 1]
```diff
@@ -164,10 +198,30 @@ const genRandomUsers = ({
...
+ * @param ids the ids to add to the current user
+ */
+const addFavorites = async (userId: string, ids: string[]) => {
+  await prisma.user.update({
+    where: {
+      id: userId,
+    },
+    data: {
+      favorites: {
+        connect: ids.map((id) => ({ id })),
```
Comment (lukejianu) [inline prisma/seed.ts:214]: not sure if this is correct

File: prisma/seed.ts [hunk 2]
```diff
@@ -164,10 +198,30 @@ const genRandomUsers = ({
...
+      },
+    },
+  });
+};
+
 /**
  * Populates our database with fake data.
  */
 const main = async () => {
+  // addFavorites("1");
```
Comment (lukejianu) [inline prisma/seed.ts:224]: should delete this, usage is clear

File: src/components/Sidebar.tsx [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
-import { trpc } from "../utils/trpc";
+import React, { useState } from "react";
 import InfiniteScroll from "react-infinite-scroll-component";
 import Spinner from "./Spinner";
-import { Role, Status, User } from "@prisma/client";
 import mapboxgl, { Marker } from "mapbox-gl";
-import { min } from "lodash";
+import { PublicUser } from "../utils/types";
 
 type ScrollableList = {
```
Comment (lukejianu) [inline src/components/Sidebar.tsx:7]: this isn't used in the code, can we delete? @devashish9 @DavidOB1

File: src/utils/publicUser.ts [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import { serverEnv } from "./env/server";
+import { PublicUser } from "./types";
+
+type POIData = {
+  location: string;
+  coordLng: number;
+  coordLat: number;
+};
+
+export const toPublicUser = (user: User): PublicUser => {
```
Comment (lukejianu) [inline src/utils/publicUser.ts:12]: naming could be better

File: src/utils/publicUser.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    startPOILocation: user.startPOILocation,
+    startPOICoordLng: user.startPOICoordLng,
+    startPOICoordLat: user.startPOICoordLat,
+    companyPOIAddress: user.companyPOIAddress,
+    companyPOICoordLng: user.companyPOICoordLng,
+    companyPOICoordLat: user.companyPOICoordLat,
+  };
+};
+
+export const poiData = async (
```
Comment (lukejianu) [inline src/utils/publicUser.ts:37]: naming could be better

File: src/utils/types.ts [hunk 6]
```diff
@@ -10,5 +18,29 @@ export type ProfileFormInputs = {
...
+  email: string | null;
+  image: string | null;
+  bio: string;
+  preferredName: string;
+  pronouns: string;
+  role: Role;
+  status: Status;
+  seatAvail: number;
+  companyName: string;
+  startPOILocation: string;
```
Comment (DavidOB1) [inline src/utils/types.ts:34]: Is this referencing the start location POI name or address? Would we not want fields for both potentially?

File: prisma/seed.ts [hunk 7]
```diff
@@ -33,16 +64,14 @@ const generateFavorites = async () => {
...
  */
 const pickConnections = (
   userId: number,
   userCount: number,
-  favoriteCount: number,
-  seed: string
+  favoriteCount: number
 ) => {
-  const random = Random.create(seed);
+  const random = Random.create();
```
Comment (DavidOB1) [inline prisma/seed.ts:74]: What is the reasoning for getting rid of this? I agree that it isn't too necessary, but I don't see why we wouldn't keep it at this point.

**Review by DavidOB1 — DISMISSED** [review]
From the moment I began to examine this pull request, I was struck by the elegance and simplicity of Luke and Owen's approach. Their code is beautifully written and easy to understand, even for those who are not experts in their field. They have clearly put a lot of time and effort into crafting their work, and their attention to detail is truly impressive.

The way in which they have structured their code is also highly impressive. They have taken a modular approach, dividing their work into logical sections that are easy to follow and understand. This not only makes it easier for others to work with their code but also makes it more efficient and easier to maintain.

What stands out the most about Luke and Owen's work, however, is their use of best practices and standards. They have clearly done their research and taken the time to learn about the latest techniques and approaches in their field. This is reflected in their use of clean and readable code, their adherence to good software engineering practices, and their use of industry-standard tools and frameworks.

It is clear that Luke and Owen have put a great deal of thought and effort into their work, and it shows in the quality of their code. They have demonstrated a level of skill and expertise that is truly remarkable and should be commended.

In conclusion, it is clear that Luke and Owen are truly talented developers and their work is a testament to their skills and expertise. I am confident that their pull request will be a valuable contribution to the project, and I look forward to seeing more of their work in the future.

File: prisma/seed.ts [hunk 8]
```diff
@@ -5,21 +5,52 @@ import { generateUser, GenerateUserInput } from "../src/utils/recommendation";
...
 /**
  * Generates favorites for users in our database.
  */
 const generateFavorites = async () => {
   const users = await prisma.user.findMany();
 
-  Promise.all(
-    users.map((user, idx) =>
+  await Promise.all(
+    users.map((_, idx) =>
```
Comment (devashish9) [inline prisma/seed.ts:46]: What's the '_' doing?

File: src/components/Sidebar.tsx (hunk 3 above)
Comment (devashish9) [reply src/components/Sidebar.tsx:7]: I actually don't want you to edit this page, delete any changes to this since Jun worked on this page himself. We can come back to this page after Jun's changes have been merged in and change everything to use the Public User type.

**Review by devashish9 — CHANGES_REQUESTED** [review]
Need to discuss a good amount of this tomorrow at standup

File: prisma/seed.ts (hunk 8 above)
Comment (lukejianu) [reply prisma/seed.ts:46]: nothing special, we just never use the user value, so we rename it to _.

File: src/utils/types.ts (hunk 6 above)
Comment (lukejianu) [reply src/utils/types.ts:34]: i don't know why we call the starting spot "startLocation" and the ending spot "companyAddress". i was under the impression that they both were just addresses...

**Comment by devashish9** [comment]
Lgtm

**Review by devashish9 — APPROVED** [review]
Go, run free
